core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -114,6 +115,8 @@ private[storage] class BlockManagerDecommissioner(
                       // driver a no longer referenced RDD with shuffle files.
                       if (bm.migratableResolver.getMigrationBlocks(shuffleBlockInfo).isEmpty) {
                         logWarning(s"Skipping block ${shuffleBlockInfo}, block deleted.")
+                      } else if (fallbackStorage.isDefined) {
+                        fallbackStorage.foreach(_.copy(shuffleBlockInfo, bm))
```

Comment:
When is the best time to move block to fallback storage during decommission? For current change, it just chooses fallback storage once encountering failure at first try.

Should we let retrying do it work until reaching `maxReplicationFailuresForDecommission` and then move to fallback storage?

Reply:
Oh I see. You only put fallback storage into peers when there is no other peer executors.

Reply:
Yes. It does.